processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -56,6 +62,18 @@
   };
   private static final byte[] SEGMENT_METADATA_CACHE_PREFIX = new byte[]{0x4};
 
+  private final SegmentMetadataQueryConfig config;
+  private final ReadablePeriod defaultHistory;
+
+  @Inject
+  public SegmentMetadataQueryQueryToolChest(
+      SegmentMetadataQueryConfig config
+  )
+  {
+    this.config = config;
```

Comment:
Storing either the config or the defaultHistory period should be enough- doesn't look like there's a need to store both